stripecardscan/src/main/java/com/stripe/android/stripecardscan/cardscan/result/MainLoopAggregator.kt

```diff
@@ -0,0 +1,95 @@
+package com.stripe.android.stripecardscan.cardscan.result
+
+import com.stripe.android.stripecardscan.cardscan.SavedFrame
+import com.stripe.android.stripecardscan.cardscan.SavedFrameType
+import com.stripe.android.stripecardscan.cardscan.CardScanConfig
+import com.stripe.android.camera.framework.AggregateResultListener
+import com.stripe.android.camera.framework.ResultAggregator
+import com.stripe.android.stripecardscan.framework.util.FrameSaver
+import com.stripe.android.stripecardscan.payment.ml.SSDOcr
+import kotlinx.coroutines.runBlocking
+
+/**
+ * Aggregate results from the main loop. Each frame will trigger an [InterimResult] to the
+ * [listener]. Once the [MainLoopState.Finished] state is reached, a [FinalResult] will be sent to
+ * the [listener].
+ *
+ * This aggregator is a state machine. The full list of possible states are subclasses of
+ * [MainLoopState].
+ */
+internal class MainLoopAggregator(
+    listener: AggregateResultListener<InterimResult, FinalResult>
+) : ResultAggregator<
+    SSDOcr.Input,
+    MainLoopState,
+    SSDOcr.Prediction,
+    MainLoopAggregator.InterimResult,
+    MainLoopAggregator.FinalResult
+    >(
+    listener = listener,
+    initialState = MainLoopState.Initial(),
+    statsName = null, // TODO: when we want to collect this in scan stats, give this a name
+) {
+
+    internal data class FinalResult(
+        val pan: String,
+        val savedFrames: Map<SavedFrameType, List<SavedFrame>>,
```

Comment:
we don't need saved frames for the scan-only flow. Saved frames are used entirely for verification purposes, so we can omit this from the final result.